### PR TITLE
Update es query source

### PR DESCRIPTION
**Pull request by TylerSheffels**
http://manage.dimagi.com/default.asp?215960

Not sure the best way to test this. I validated that the call that was failing on ES now succeeds with the generated query and added a test to ESQuery, so I feel pretty good about it.

@snopoke

### CODE REVIEW

File: corehq/apps/es/tests.py [hunk 1]
```diff
@@ -358,3 +358,24 @@ def test_nested_filter(self):
...
+                    },
+                    "query": {"match_all": {}}
+                }
+            },
+            "size": SIZE_LIMIT,
+            "_source": ["source_obj"]
+        }
+        q = HQESQuery('forms').source('source_obj')
+        self.checkQuery(q, json_output)
+
```
Comment (dimagimon) [inline corehq/apps/es/tests.py]: W391 blank line at end of file

**Comment by snopoke** [comment]
looks good, I tested it via ES head and seems like it should work

File: corehq/apps/es/es_query.py [hunk 2]
```diff
@@ -244,13 +246,35 @@ def _assemble(self):
...
+            self.es_query['_source'] = self._source
 
     def fields(self, fields):
-        """Restrict the fields returned from elasticsearch"""
+        """
+            Restrict the fields returned from elasticsearch
+
+            Usage Note: As of ES 1.x, fields will only work on leaf nodes! It will no longer return an object,
+            e.g. field.*, to return an object refer to '.source'
+            """
```
Comment (esoergel) [inline corehq/apps/es/es_query.py]: Should we just switch everything to use `_source`?  The [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-fields.html) make it sound like that's the preferred approach (see warning at the top).  It also feels more in-line with our usage.  @snopoke thoughts?

Comment (snopoke) [inline corehq/apps/es/es_query.py]: Seems like more closely match how we use it.

**Pull request merged**